src/LuaPiGui.cpp

```diff
@@ -16,42 +16,43 @@
 #include "SystemInfoView.h"
 #include "WorldView.h"
 #include "graphics/Graphics.h"
+#include "pigui/LuaFlags.h"
 #include "ship/PlayerShipController.h"
 #include "sound/Sound.h"
 #include "ui/Context.h"
 
-#include <numeric>
 #include <iterator>
+#include <numeric>
 
 // Windows defines RegisterClass as a macro, but we don't need that here.
 // undef it, to avoid including yet another header that undefs it
 #undef RegisterClass
 
 template <typename Type>
-static Type parse_imgui_flags(lua_State *l, int index, std::map<std::string, Type> table, std::string name)
+static Type parse_imgui_flags(lua_State *l, int index, LuaFlags<Type> &lookupTable)
 {
 	PROFILE_SCOPED()
-	LuaTable flags(l, index);
-	Type theflags = Type(0);
-	for (LuaTable::VecIter<std::string> iter = flags.Begin<std::string>(); iter != flags.End<std::string>(); ++iter) {
-		std::string flag = *iter;
-		if (table.find(flag) != table.end())
-			theflags = static_cast<Type>(theflags | table.at(flag));
-		else
-			Error("Unknown %s %s\n", name.c_str(), flag.c_str());
+	Type theFlags = Type(0);
+	if (lua_isnumber(l, index)) {
+		theFlags = static_cast<Type>(lua_tointeger(l, index));
+	} else if (lua_istable(l, index)) {
+		theFlags = lookupTable.LookupTable(l, index);
+	} else {
+		Error("Expected a table or integer, got %s.\n", luaL_typename(l, index));
```

Comment:
Would be a good idea to add `luaL_traceback` here, and in all the other parsing functions, to the error output, hunting down the source of the error without it is a bit tedious without it.